### PR TITLE
[release/3.1] Align CODEOWNERS with 'main'

### PR DESCRIPTION
**Pull request by dougbu**
- add and remove many assignments

This is general cleanup but I added a few assignments too.

### CODE REVIEW

**Comment by dougbu** [comment]
I _think_ I requested reviews from most affected people. Feel free to /cc or add reviewers if I missed someone.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,22 +1,25 @@
...
+/eng/                           @dotnet/aspnet-build
 /eng/common/                    @dotnet-maestro-bot
 /eng/Versions.props             @dotnet-maestro-bot @dougbu
 /eng/Version.Details.xml        @dotnet-maestro-bot @dougbu
-/src/Components/                @SteveSandersonMS
-/src/DefaultBuilder/            @tratcher @anurse
-/src/Hosting/                   @tratcher @anurse
-/src/Http/                      @tratcher @jkotalik @anurse
-/src/Middleware/                @tratcher @anurse
-/src/ProjectTemplates/          @ryanbrandenburg
```
Comment (dougbu) [inline .github/CODEOWNERS:17]: We don't have an assignment for this subfolder or lots of /src/ here or in 'main'. Do we need more ownerships❔

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -1,22 +1,25 @@
 # Users referenced in this file will automatically be requested as reviewers for PRs that modify the given paths.
 # See https://help.github.com/articles/about-code-owners/
 
-/global.json                    @aspnet/build
```
Comment (dougbu) [inline .github/CODEOWNERS:4]: @Pilchie do you want to own everything as you do in 'release/6.0' and 'main'❔

**Comment by dougbu** [comment]
Will rerun once #35637 is in

File: .github/CODEOWNERS (hunk 2 above)
Comment (Pilchie) [reply .github/CODEOWNERS:4]: > @Pilchie do you want to own everything as you do in 'release/6.0' and 'main'❔

Sure.

**Review by Tratcher — APPROVED** [review]

**Review by wtgodbe — APPROVED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (wtgodbe) [reply .github/CODEOWNERS:17]: Yeah, we should address that. @pranavkm for this one?

**Pull request merged**